src/Control/Distributed/Process/Internal/Types.hs

```diff
@@ -497,6 +511,10 @@ data DiedReason =
   | DiedUnknownId
   deriving (Show, Eq)
 
+instance NFData DiedReason where
+  rnf (DiedException s) = rnf s `seq` ()
```

Comment:
No, autoderiving creates incorrect instances in <GHC-7.10. Those instances will be evaluated to WHNF, not NF.